services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/CredentialsDto.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Creates a new data transfer object to store credentials information in mongodb.
+     *
+     * @param tenantId The tenant identifier.
+     * @param deviceId The device identifier.
+     * @param credentials The list of credentials.
+     * @param version The version of tenant to be sent as request header.
+     * @throws NullPointerException if any of the parameters except the secrets are {@code null}
+     * @throws IllegalArgumentException if validation of the given credentials fail.
+     */
+    public CredentialsDto(
+            final String tenantId,
+            final String deviceId,
+            final List<CommonCredential> credentials,
+            final String version) {
+
+        setTenantId(tenantId);
+        setDeviceId(deviceId);
+
+        //Validate the given credentials, secrets and generate secret ids if not available.
+        Optional.ofNullable(credentials)
+                .ifPresent(ok -> {
+                    validateForUniqueAuthIdAndType(credentials);
+                    validateSecretsAndGenerateIds(credentials);
+                });
+        setCredentials(credentials);
+
+        setVersion(version);
+        setUpdatedOn(Instant.now());
+    }
+
+    /**
+     * Gets the identifier of the tenant.
+     *
+     * @return The identifier of the tenant.
+     */
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    /**
+     * Sets the identifier of the tenant.
+     *
+     * @param tenantId The tenant's identifier.
+     * @throws NullPointerException if the tenantId is {@code null}.
+     */
+    public void setTenantId(final String tenantId) {
+        this.tenantId = Objects.requireNonNull(tenantId);
+    }
+
+    /**
+     * Gets the identifier of the device.
+     *
+     * @return The identifier of the device.
+     */
+    public String getDeviceId() {
+        return deviceId;
+    }
+
+    /**
+     * Sets the identifier of the device.
+     *
+     * @param deviceId The identifier of the device.
+     * @throws NullPointerException if the deviceId is {@code null}.
+     */
+    public void setDeviceId(final String deviceId) {
+        this.deviceId = Objects.requireNonNull(deviceId);
+    }
+
+    /**
+     * Gets the list of credentials.
+     *
+     * @return The list of credentials.
+     */
+    public List<CommonCredential> getCredentials() {
+        return credentials;
+    }
+
+    /**
+     * Sets the list of credentials.
+     *
+     * @param credentials A list of credentials.
+     */
+    public void setCredentials(final List<CommonCredential> credentials) {
+        this.credentials = credentials;
+    }
+
+    /**
+     * Checks whether the credentials to be updated already have secret ids.
+     *
+     * @return hasSecretIds {@code true} if any credential already has a secret identifier,
+     *                      otherwise {@code false}
+     */
+    @JsonIgnore
+    public boolean hasSecretIds() {
+        return hasSecretIds;
+    }
```

Comment:
I'd rather check the (few) secrets explicitly ...

Reply:
I didn't get it what do you mean, could you please explain more detailed? I set this flag, if the secrets to be updated contain any secret ids. I use this flag during the update operation, if no secret ids exist then no need to merge and I simply replace the existing document.